cmd/snap/cmd_run_test.go

```diff
@@ -499,3 +501,70 @@ func (s *SnapSuite) TestSnapRunAppIntegrationFromCore(c *check.C) {
 		"snapname.app", "--arg1", "arg2"})
 	c.Check(execEnv, testutil.Contains, "SNAP_REVISION=x2")
 }
+
+func (s *SnapSuite) TestSnapRunXauthorityMigration(c *check.C) {
+	// mock installed snap; happily this also gives us a directory
+	// below /tmp which the Xauthority migration expects.
+	dirs.SetRootDir(c.MkDir())
+	defer func() { dirs.SetRootDir("/") }()
+	defer mockSnapConfine()()
+
+	u, err := user.Current()
+	c.Assert(err, check.IsNil)
+
+	// Ensure XDG_RUNTIME_DIR exists for the user we're testing with
+	os.MkdirAll(filepath.Join(dirs.XdgRuntimeDirBase, u.Uid), 0700)
+
+	si := snaptest.MockSnap(c, string(mockYaml), string(mockContents), &snap.SideInfo{
+		Revision: snap.R("x2"),
+	})
+	err = os.Symlink(si.MountDir(), filepath.Join(si.MountDir(), "../current"))
+	c.Assert(err, check.IsNil)
+
+	// redirect exec
+	execArg0 := ""
+	execArgs := []string{}
+	execEnv := []string{}
+	restorer := snaprun.MockSyscallExec(func(arg0 string, args []string, envv []string) error {
+		execArg0 = arg0
+		execArgs = args
+		execEnv = envv
+		return nil
+	})
+	defer restorer()
+
+	// Ensure we have XDG_RUNTIME_DIR for the current user
+	os.MkdirAll(filepath.Join(dirs.XdgRuntimeDirBase, u.Uid), 0700)
```

Comment:
Is that the same os.MkdirAll() as a few lines above? Or a different one? If a different one, maybe a small comment whats different about it?